apidef/oas/security.go

```diff
@@ -71,14 +70,18 @@ func (s *OAS) extractSecurityTo(api *apidef.APIDefinition) {
 		api.AuthConfigs = make(map[string]apidef.AuthConfig)
 	}
 
-	if len(s.Security) == 0 {
+	if len(s.Security) == 0 || len(s.Components.SecuritySchemes) == 0 {
 		return
 	}
 
-	for name := range s.Security[0] {
-		switch s.Components.SecuritySchemes[name].Value.Type {
-		case apiKey:
-			s.extractTokenTo(api, name)
+	for schemeName := range s.getTykSecuritySchemes() {
+		if _, ok := s.Security[0][schemeName]; ok {
+			switch s.Components.SecuritySchemes[schemeName].Value.Type {
```

Comment:
switch + single case:, perhaps convert to if? (gosimple)

Reply:
The switch will have more cases as we add other authentication modes.